### ISSUE TITLE
Question: Is it possible to reject a subject during simulation?

### ISSUE THREAD

**Issue by billdenney** [nlmixrdevelopment/nlmixr#506]
I am trying to simulate a clinical trial with inclusion criteria that a lab measurement must be >x.  For simplicity of this discussion, let's say that hemoglobin must be >10 g/dL.

In the ideal scenario, I would like to be able to do all of the following within the simulation model:
* Simulate the first time point (to be considered like screening)
* Within that first time point, use the simulated value with both inter-individual variability (IIV) and residual variability to assess if the subject passes screening (note that this is not just IIV because I do anticipate some subjects to enter the trial based on residual variability when they would not enter the trial based on IIV-alone)
* If the subject passes screening test, continue for a full simulation.  If the subject does not pass the screening test, select a new subject (new IIV and new residuals, and run again).

The simple-enough work-around is to simulate more subjects than required, run the full simulation for all subjects, and only use the first N who pass that screening criteria.  In other words, do that filtering on the simulated results without any in-model filtering.

... as I'm thinking more about this while writing the issue, I will likely do the "simple-enough work-around", but I remain curious if a method exists to do the above.

**Comment by mattfidler** [comment]
Yes it is possible. Since it isn't pressing I will get an example later.

**Comment by mattfidler** [comment]
You can use `while` and `simeps()`, `simeta()` to reset the individual, but looking more carefully at the problem you are trying to simulate and reject after running the ODE solving for 1 or 2 samples, which is not supported currently.  It could be supported in RxODE but not yet.
